codegen/tests/simple/app.go

```diff
@@ -0,0 +1,37 @@
+// Code generated by sysl DO NOT EDIT.
+package simple
+
+import (
+	"github.com/anz-bank/sysl-go/codegen/tests/deps"
+	"github.com/anz-bank/sysl-go/codegen/tests/downstream"
+	"github.com/anz-bank/sysl-go/config"
+	"github.com/anz-bank/sysl-go/core"
+	"github.com/anz-bank/sysl-go/handlerinitialiser"
+)
+
+// HandlerInit for Simple
+type HandlerInit struct {
+	enabledHandlers     []handlerinitialiser.HandlerInitialiser
+	enabledGrpcHandlers []handlerinitialiser.GrpcHandlerInitialiser
+}
+
+// InitialiseHandler ...
+func InitialiseHandler(coreCfg *config.DefaultConfig, serviceInterface ServiceInterface, serviceCallback core.RestGenCallback) (*HandlerInit, error) {
+	var err error = nil
+	depsHTTPClient, depsErr := core.BuildDownstreamHTTPClient("deps", &coreCfg.GenCode.Downstream.(*DownstreamConfig).Deps)
+	downstreamHTTPClient, downstreamErr := core.BuildDownstreamHTTPClient("downstream", &coreCfg.GenCode.Downstream.(*DownstreamConfig).Downstream)
+	if depsErr != nil {
+		return nil, depsErr
+	}
+
+	if downstreamErr != nil {
+		return nil, downstreamErr
+	}
+
+	depsClient := deps.NewClient(depsHTTPClient, coreCfg.GenCode.Downstream.(*DownstreamConfig).Deps.ServiceURL)
+	downstreamClient := downstream.NewClient(downstreamHTTPClient, coreCfg.GenCode.Downstream.(*DownstreamConfig).Downstream.ServiceURL)
+	serviceHandler := NewServiceHandler(serviceCallback, &serviceInterface, depsClient, downstreamClient)
+	serviceRouter := NewServiceRouter(serviceCallback, serviceHandler)
+	httpHandlers := []handlerinitialiser.HandlerInitialiser{serviceRouter}
+	return &HandlerInit{enabledHandlers: httpHandlers}, err
```

Comment:
The types are fine for now. Some thought will need to be in to supporting both REST and GRPC soon.